src/scripts/main.js

```diff
@@ -1,3 +1,34 @@
 'use strict';
 
-// write code here
+const receivedList = document.querySelector('ul');
+
+function parseSalary(salaryString) {
+  return Number(salaryString.replace(/[$,]/g, ''));
+}
+
+function sortList(list) {
+  const listItems = Array.from(list.children);
+
+  listItems.sort((a, b) => {
+    const salaryA = parseSalary(a.dataset.salary);
+    const salaryB = parseSalary(b.dataset.salary);
+
+    return salaryB - salaryA;
+  });
+
+  listItems.forEach((item) => {
+    return list.appendChild(item);
```

Comment:
Using `appendChild` inside a `forEach` loop directly on the list can cause issues if the list is being manipulated during iteration. Consider using a `documentFragment` to append all items first and then append the fragment to the list for better performance and to avoid potential reflow issues.